File: example-jpms/src/test/java/example/jpms/AggregateFunctionTest.java
package example.jpms;

import static org.junit.jupiter.api.Assertions.assertEquals;

import example.jpms.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class AggregateFunctionTest {

  private final EmployeeRepository repository;

  AggregateFunctionTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testCount() {
    assertEquals(14L, repository.selectCount());
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/AggregateFunctionTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class AggregateFunctionTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testCount() {
        assertEquals(14L, repository.selectCount())
    }
}
